package/gluon-mesh-vpn-tunneldigger/files/usr/bin/tunneldigger-watchdog

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/lua
+
+function restart_tunneldigger()
+	io.popen('logger -t tunneldigger-watchdog "Restarting Tunneldigger."')
+	os.execute('/etc/init.d/tunneldigger restart')
+end
+
+function read_pid_file()
+	local pid_file = io.open('/var/run/tunneldigger.mesh-vpn.pid', 'r')
+	if not pid_file then
+		return nil
+	end
+	local pid = pid_file:read('*l') 
+	pid_file:close()
+	return pid
+end
+
+function has_mesh_vpn_neighbours()
+	local handle = io.popen('batctl o', 'r')
+	if handle then
+		for line in handle:lines() do
+			if line:find('mesh%-vpn') then
+				handle:close()
+				return true
+			end
+		end
+	end
+	handle:close()
```

Comment:
You can now end up calling `close` on a `nil` value here.

Structure it like

```
if not handle then
  return false
end
for line in handle:lines() do
  .....
end
handle:close()
return false
```

Reply:
✓